src/app/pages/component-viewer/component-viewer.ts

```diff
@@ -63,47 +63,74 @@ export class ComponentViewer implements OnDestroy {
   }
 }
 
-@Component({
-  selector: 'component-overview',
-  templateUrl: './component-overview.html',
-  encapsulation: ViewEncapsulation.None,
-})
-export class ComponentOverview implements OnInit {
+export class ComponentBaseView implements OnInit, OnDestroy {
   @ViewChild('initialFocusTarget', {static: false}) focusTarget: ElementRef;
   @ViewChild('toc', {static: false}) tableOfContents: TableOfContents;
+
   showToc: Observable<boolean>;
 
+  destroyed = new Subject<void>();
+
   constructor(public componentViewer: ComponentViewer, breakpointObserver: BreakpointObserver) {
     this.showToc = breakpointObserver.observe('(max-width: 1200px)')
       .pipe(map(result => !result.matches));
   }
 
   ngOnInit() {
-    // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
-    setTimeout(() => this.focusTarget.nativeElement.focus({preventScroll: true}), 100);
+    this.componentViewer.componentDocItem.pipe(takeUntil(this.destroyed)).subscribe(item => {
+      // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
+      setTimeout(() => this.focusTarget.nativeElement.focus({preventScroll: true}), 100);
+      if (this.tableOfContents) {
+        this.tableOfContents.resetHeaders();
+      }
+    });
+  }
+
+  ngOnDestroy() {
+    this.destroyed.next();
   }
 
-  scrollToSelectedContentSection() {
+  onContentRendered(sectionName: string, docViewerContent: HTMLElement) {
```

Comment:
Why change this name? I prefer naming functions based on what they do rather than when they're called

Reply:
Changed since its scope of responsibility broadened, but since it's purpose is to make sure the Table of Contents is accurate, I can change to `updateTableOfContents`.